scripts/tests/run_tv_casting_test.py

```diff
@@ -0,0 +1,305 @@
+
+
+class LogFileManager:
+    """
+    A context manager for managing log files.
+
+    This class provides a context manager for safely opening and closing log files.
+    It ensures that log files are properly managed, allowing reading, writing, or 
+    both, depending on the specified mode.
+    """
+
+    # Initialize LogFileManager.
+    #   log_file_path (str): The path to the log file.
+    #   mode (str): The file mode for opening the log file (default is 'w+' for read/write mode).
+    def __init__(self, log_file_path: str, mode: str = 'w+'):
+        self.log_file_path = log_file_path
+        self.mode = mode
+
+    # Enter the context manager to open and return the log file.
+    def __enter__(self):
+        try:
+            self.file = open(self.log_file_path, self.mode)
+        except FileNotFoundError:
+            # Handle file not found error
+            raise FileNotFoundError(f"Log file '{self.log_file_path}' not found.")
+        except IOError:
+            # Handle IO error
+            raise IOError(f"Error opening log file '{self.log_file_path}'.")
+        return self.file
+
+    # Exit the context manager, closing and removing the log file created.
+    #   exception_type: The type of exception that occurred, if any.
+    #   exception_value: The value of the exception, if any.
+    #   traceback: The traceback of the exception.
+    def __exit__(self, exception_type, exception_value, traceback):
+        self.file.close()
+
+        if os.path.exists(self.log_file_path):
+            os.remove(self.log_file_path)
+
+
+class ProcessManager:
+    """
+    A context manager for managing subprocesses.
+
+    This class provides a context manager for safely starting and stopping a subprocess.
+    """
+
+    # Initialize ProcessManager.
+    #   command (list): The command to execute as a subprocess.
+    #   stdout (file): File-like object to which the subprocess's standard output will be redirected.
+    #   stderr (file): File-like object to which the subprocess's standard error will be redirected.
+    def __init__(self, command: List[str], stdout, stderr):
+        self.command = command
+        self.stdout = stdout
+        self.stderr = stderr
+
+    # Enter the context manager to start the subprocess and return it.
+    def __enter__(self):
+        self.process = subprocess.Popen(self.command, stdout=self.stdout, stderr=self.stderr, text=True)
+        return self.process
+
+    # Exit the context manager, terminating the subprocess.
+    #   exception_type: The type of exception that occurred, if any.
+    #   exception_value: The value of the exception, if any.
+    #   traceback: The traceback of the exception.
+    def __exit__(self, exception_type, exception_value, traceback):
+        self.process.terminate()
+        self.process.wait()
+
+
+# Dump the contents of a log file to the console.
+#   log_file_path: The path to the log file.
+def dump_logs_to_console(log_file_path: str):
```

Comment:
Consider adding information that passed in logs are removed once processed. That may not be intuitive. Something like `dump_temporary_logs_to_console` and comment should say that the log file will get removed once the function exits.

Reply:
You seem to already use LogFileManager inside the main test ... in that case does this still need to be a log file manager that removes files? Maybe yes because sys.exit, however then maybe add a comment explaining this.

Reply:
This function name has been updated per the suggestion.

As noted in another comment and reply, the LogFileManger has been removed as the temporary directory takes care of the file clean up.